packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/InstanceManager.java

```diff
@@ -192,26 +174,23 @@ public <T> T getInstance(long identifier) {
    * Returns whether this manager contains the given `instance`.
    *
    * @param instance the instance whose presence in this manager is to be tested.
-   * @return whether this manager contains the given `instance`. If the manager is closed, returns
-   *     `false`.
+   * @return whether this manager contains the given `instance`.
    */
   public boolean containsInstance(Object instance) {
-    if (isClosed()) {
-      Log.w(TAG, CLOSED_WARNING);
-      return false;
-    }
+    logWarningIfFinalizationListenerHasStopped();
     return identifiers.containsKey(instance);
   }
 
   /**
-   * Closes the manager and releases resources.
+   * Stop the periodic run of the {@link FinalizationListener} for instances that have been garbage
+   * collected.
    *
-   * <p>Methods called after this one will be ignored and log a warning.
+   * <p>The InstanceManager can continue to be used, but the {@link FinalizationListener} will no
+   * longer be called and methods will log a warning.
    */
-  public void close() {
+  public void stopFinalizationListener() {
     handler.removeCallbacks(this::releaseAllFinalizedInstances);
-    isClosed = true;
-    clear();
+    hasFinalizationListenerStopped = true;
```

Comment:
Just to make sure I understand correctly, can you confirm the following is correct?

After this method called, objects that have been added to this instance manager can be garbage collected. Thus, objects might be contained by the instance manager (because we aren't releasing the resources/calling `clear()` anymore), but also might not. So basically, the instance manager is now unreliable, but with this change, it's up the developer to be aware of that versus getting a null, -1, etc. value back when attempting to call a method?

Reply:
When a new instance is added to the manager, the manager creates a strong reference and a weak reference to the instance. When `remove(Object)` is called, the strong reference is removed. Therefore, the weak reference to the instance will remain until it is garbage collected. The manager now functions this way whether this value is `true` or `false`. The only difference now is that the `finalizationListener` that was passed in `InstanceManager.create`  is no longer called when the weak reference to an instance is claimed by garbage collection.

Reply:
Is it possible to test the case where the weak reference was claimed by garbage collection to see that the `finalizationListener` is no longer called? I figure it's not but asking just in case!

Reply:
It is possible with the `Runtime.getRuntime().gc();` method. I tried to add a unit test, but the Android `Handler` class is just a stub when running unit tests. So I created an integration test.